src/main/java/frc/robot/subsystems/GBSubsystem.java

```diff
@@ -1,32 +1,45 @@
 package frc.robot.subsystems;
 
 import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.InstantCommand;
+import edu.wpi.first.wpilibj2.command.SequentialCommandGroup;
 import edu.wpi.first.wpilibj2.command.SubsystemBase;
 import org.littletonrobotics.junction.Logger;
 
 public abstract class GBSubsystem extends SubsystemBase {
 
 	private final String logPath;
+	private String currentCommandName;
 
 	public GBSubsystem(String logPath) {
 		this.logPath = logPath;
+		this.currentCommandName = "No command is currently running on the"
 	}
-
-	private String getCurrentCommandName() {
-		Command currentCommand = getCurrentCommand();
-		return currentCommand != null ? currentCommand.getName() : "no command is currently running on the subsystem";
-	}
-
+	
 	public String getLogPath() {
 		return logPath;
 	}
 
 	@Override
 	public final void periodic() {
-		Logger.recordOutput(getLogPath() + "CurrentCommand", getCurrentCommandName());
+		Logger.recordOutput(getLogPath() + "CurrentCommand", currentCommandName);
 		subsystemPeriodic();
 	}
 
 	protected void subsystemPeriodic() {}
 
+	protected Command addNameToCommand(Command command){
+		return new SequentialCommandGroup(
+				new InstantCommand(() -> currentCommandName = command.getName()),
+				command
+		);
```

Comment:
but it doesnt solve the problem if u still use command.getName(). the purpose of the projects to not use it